targets/TARGET_STM/sleep.c

```diff
@@ -83,16 +177,21 @@ void hal_deepsleep(void)
 #else /* TARGET_STM32L4 */
     HAL_PWR_EnterSTOPMode(PWR_LOWPOWERREGULATOR_ON, PWR_STOPENTRY_WFI);
 #endif /* TARGET_STM32L4 */
+    // Verify Clock Out of Deep Sleep
+    ForceClockOutofDeepSleep();
 
     // Restart HAL tick
     HAL_ResumeTick();
 
-    // Enable IRQs
-    core_util_critical_section_exit();
-
     // After wake-up from STOP reconfigure the PLL
     SetSysClock();
 
+    /*  Wait for clock to be stabilized.
+     *  TO DO: a better way of doing this, would be to rely on
+     *  HW Flag. At least this ensures proper operation out of
+     *  deep sleep */
+    wait_loop(500);
```

Comment:
is there any hw flag actually, what hw are we waiting here?

500 us might a lot in some cases?

Reply:
It takes around 1 to 2ms to settle the PLLs. So 0.5ms is in the same range. I'd prefer to avoid it and find a better way (HW flag ..) butI haven't found it so far ... but ST internal discussions are ongoing